dlp/inspectWithCustomRegex.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//  title: Inspects strings
+//  description: Inspects a string using custom regex pattern.
+//  usage: node inspectWithCustomRegex.js my-project string minLikelihood maxFindings infoTypes customInfoTypes includeQuote
```

Comment:
Can we trim this down?

- `string`: required
- `minLikelihood`: we should be able to remove this because the likelihood for custom infotype (which is how regex is used) is controlled by the request anyway.
- `maxFindings`: does not make sense in the context of this sample. We're inspecting a small-ish string, not megabytes of file content, so there will only be so many findings. If you want to limit it anyway, just hardcode a limit of ~1000 in the sample. Users can change it if they need to.
- `infoTypes`: Can omit, since we're demonstrating regex matching only. I guess it's possible we may want to show side-by-side detection of custom and built-in infotypes, but if that's the case move this to the end and make it optional. (Also if that is the case, lets make the example string actually demonstrate that)
- `customInfoTypes`: Since the whole point of this sample is to demonstrate regex, we should ask for regex directly and construct the custom infotype in code.
- `includeQuote`: as with maxFindings, lets just set this to true for demo purposes. If users want to change it they can edit the code.

At a high level we should make the sample as easy as possible to run. Adding a lot of parameters and using obscure syntax (such as the ',' and regex/dict hybrid for customInfoTypes) will lead to confusion and frustration.

As a user of this sample, I should be able to say `node inspectWithCustomRegex.js 'this is my serial number aab-bcdd-eef' '[a-f\-]10'` and see results.

Reply:
@soumya92 I had the same thought when I started the implementation but I noticed a particular structure is being followed for all inspect samples. Couldn't figure out the exact reason but mostly it was to keep the sample code consistent. Anyway, I feel your findings look reasonable and so I have updated this sample. Also, will it be okay if I make these same changes in my other PRs?

Reply:
Yeah go for it! The easier we make our samples to use, the better